### ISSUE TITLE
Uncaught TypeError: Cannot read properties of undefined (reading 'neighbors')

### ISSUE THREAD

**Issue by zepumph** [phetsims/density-buoyancy-common#390]
Hit this while testing in https://github.com/phetsims/density-buoyancy-common/issues/369:

http://localhost:8080/buoyancy/buoyancy_en.html?brand=phet-io&phetioStandalone&ea&debugger&preferencesStorage&fuzz&fuzzBoard&disableModals

I don't think we need to do much here, just want to track it in case it starts showing up more. 

```
p2-0.7.1.js:12276 Uncaught TypeError: Cannot read properties of undefined (reading 'neighbors')
    at IslandManager.split (p2-0.7.1.js:12276:12)
    at World.internalStep (p2-0.7.1.js:13093:27)
    at World.step (p2-0.7.1.js:12895:18)
    at PhysicsEngine.step (PhysicsEngine.ts:119:16)
    at BuoyancyShapesModel.step (DensityBuoyancyModel.ts:398:17)
    at Sim.ts:444:22
    at PhetioAction.execute (PhetioAction.ts:162:17)
    at Sim.stepSimulation (Sim.ts:1068:31)
    at Sim.stepOneFrame (Sim.ts:1058:12)
    at Sim.runAnimationLoop (Sim.ts:1033:12)
```

**Comment by samreid** [comment]
Relevant code in the area:

```ts
    // Add connectivity data. Each equation connects 2 bodies.
    for(var k=0; k!==equations.length; k++){
        var eq=equations[k],
            i=bodies.indexOf(eq.bodyA),
            j=bodies.indexOf(eq.bodyB),
            ni=nodes[i],
            nj=nodes[j];
        ni.neighbors.push(nj);
        nj.neighbors.push(ni);
        ni.equations.push(eq);
        nj.equations.push(eq);
    }
```

**Comment by zepumph** [comment]
I believe we have a reproducible case over in https://github.com/phetsims/density-buoyancy-common/issues/399#issuecomment-2347250452:

>After Reset All, if you try to select "same volume" or a different shape on the shapes screen with the mouse this console error repeatedly pops up:
><img width="364" alt="Screenshot 2024-09-12 at 5 07 53 PM" src="https://github.com/user-attachments/assets/feed401d-8c9e-457e-8b65-4292ab0381e4">

**Comment by zepumph** [comment]
I was not able to reproduce this as easily as I thought I'd be able to using the above steps. @samreid can you?

**Comment by samreid** [comment]
I investigated the shapes screen and compare screen, changing shapes and radio buttons and testing with reset all. I could not trigger any errors.

We could try to be defensive and try/catch the model step? But it could leave things in an awkward or state, or repeatedly error over and over:

<details>

```diff

Subject: [PATCH] Add notes
---
Index: js/common/model/PhysicsEngine.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/js/common/model/PhysicsEngine.ts b/js/common/model/PhysicsEngine.ts
--- a/js/common/model/PhysicsEngine.ts	(revision 8773c47993274ad5ca0a346632f28eae0c8788cf)
+++ b/js/common/model/PhysicsEngine.ts	(date 1728053705211)
@@ -116,8 +116,13 @@
    * per simulation step.
    */
   public step( dt: number ): void {
-    this.world.step( FIXED_TIME_STEP, dt, MAX_SUB_STEPS );
-    this.interpolationRatio = ( this.world.accumulator % FIXED_TIME_STEP ) / FIXED_TIME_STEP;
+    try {
+      this.world.step( FIXED_TIME_STEP, dt, MAX_SUB_STEPS );
+      this.interpolationRatio = ( this.world.accumulator % FIXED_TIME_STEP ) / FIXED_TIME_STEP;
+    }
+    catch( e ) {
+      console.log( e );
+    }
   }
 
   /**

**Comment by samreid** [comment]
@zepumph and I do not want to add a workaround or silence the error. We decided to mark as ready for review, and to ask @Nancy-Salpepi and @KatieWoe to see if it occurs at all during RC.2.  If not, this issue can be closed.

**Comment by Nancy-Salpepi** [comment]
Since alt input isn't working correctly over in https://github.com/phetsims/density-buoyancy-common/issues/399, I am unable to completely test this issue for https://github.com/phetsims/qa/issues/1149

**Comment by samreid** [comment]
We corrected #399 and this will be ready for a look in RC.3

**Comment by Nancy-Salpepi** [comment]
This looks good to me in rc.3. Closing
